src/main/java/net/rubygrapefruit/platform/internal/jni/AbstractFileEventFunctions.java

```diff
@@ -66,33 +179,62 @@ public NativeFileWatcherCallback(BlockingQueue<FileWatchEvent> eventQueue) {
         // Called from the native side
         @SuppressWarnings("unused")
         public void pathChanged(int typeIndex, String path) {
-            FileWatchEvent.Type type = FileWatchEvent.Type.values()[typeIndex];
-            if (type == OVERFLOWED) {
-                signalOverflow(path);
-            } else {
-                queueEvent(new ChangeEvent(type, path), false);
+            EventType type = EventType.values()[typeIndex];
+            switch (type) {
+                case OVERFLOWED:
+                    signalOverflow(OverflowType.OPERATING_SYSTEM, path);
```

Comment:
Can we do the distinction here already on the native side, i.e. the native side calls different methods for the different `switch` branches here?

Reply:
I did that.

Reply:
> I did that

You did what? There is no separate callback for the overflow and unknown events from the native side.

Reply:
Looks like GitHub lost some of my changes. Fuck.